From 7550077e243e8a5a6ad26bf8956101ca9259dfcf Mon Sep 17 00:00:00 2001
From: Heinrich Schuchardt <xypron.glpk@gmx.de>
Date: Tue, 24 Oct 2017 18:07:17 +0200
Subject: [PATCH 1/1] efi_loader: efi_net: check return value of calloc

Calloc may return NULL. So we must check the return value.

Signed-off-by: Heinrich Schuchardt <xypron.glpk@gmx.de>
---
 lib/efi_loader/efi_net.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/lib/efi_loader/efi_net.c b/lib/efi_loader/efi_net.c
index 432d9a99a2..a7b101e830 100644
--- a/lib/efi_loader/efi_net.c
+++ b/lib/efi_loader/efi_net.c
@@ -292,6 +292,10 @@ int efi_net_register(void)
 
 	/* We only expose the "active" eth device, so one is enough */
 	netobj = calloc(1, sizeof(*netobj));
+	if (!netobj) {
+		printf("ERROR: Out of memory\n");
+		return 1;
+	}
 
 	/* Fill in object data */
 	netobj->parent.protocols[0].guid = &efi_net_guid;
-- 
2.11.0

